### PR TITLE
[13.0][WIP][IMP] sale_payment_sheet: Show invoices in wizard payments and improve inheritance

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT33281

ping @carlosdauden @chienandalu

### CODE REVIEW

File: sale_payment_sheet/wizards/sale_invoice_payment.py [hunk 1]
```diff
@@ -23,21 +23,37 @@ class SaleInvoicePaymentWiz(models.TransientModel):
...
+        required=True,
+        compute="_compute_amount",
+        readonly=False,
+        store=True,
     )
     ref = fields.Char(string="Reference")
+    invoice_ids = fields.Many2many(comodel_name="account.move",)
+    partner_id = fields.Many2one(comodel_name="res.partner")
+
+    @api.depends("invoice_ids")
```
Comment (cesar-tecnativa) [inline sale_payment_sheet/wizards/sale_invoice_payment.py:37]: '    @api.depends("invoice_ids", "invoice_ids.type", "invoice_ids.amount_residual")

File: sale_payment_sheet/wizards/sale_invoice_payment.py [hunk 2]
```diff
@@ -23,21 +23,37 @@ class SaleInvoicePaymentWiz(models.TransientModel):
...
+    def _compute_amount(self):
+        for wiz in self:
+            amount = 0.0
+            for invoice in wiz.invoice_ids:
+                if invoice.type == "out_refund":
```
Comment (cesar-tecnativa) [inline sale_payment_sheet/wizards/sale_invoice_payment.py:42]: if invoice.type in ('out_invoice', 'out_refund'):

File: sale_payment_sheet/wizards/sale_invoice_payment.py [hunk 3]
```diff
@@ -23,21 +23,37 @@ class SaleInvoicePaymentWiz(models.TransientModel):
...
         res = super(SaleInvoicePaymentWiz, self).default_get(fields_list)
-        invoices = self.env["account.move"].browse(self.env.context.get("active_ids"))
         res["journal_id"] = self.env.user.commercial_journal_ids[:1].id
-        res["amount"] = 0.0
-        for invoice in invoices:
-            if invoice.type == "out_refund":
-                res["amount"] -= invoice.amount_residual
-            else:
-                res["amount"] += invoice.amount_residual
+        if self.env.context.get("active_model", False) != "account.move":
```
Comment (cesar-tecnativa) [inline sale_payment_sheet/wizards/sale_invoice_payment.py:52]: if self.env.context.get("active_model", '') != "account.move":

cannot compare boolean with strings

File: sale_payment_sheet/wizards/sale_invoice_payment.py [hunk 4]
```diff
@@ -23,21 +23,37 @@ class SaleInvoicePaymentWiz(models.TransientModel):
...
+        currency_field="currency_id",
+        string="Amount",
+        required=True,
+        compute="_compute_amount",
+        readonly=False,
+        store=True,
     )
     ref = fields.Char(string="Reference")
+    invoice_ids = fields.Many2many(comodel_name="account.move",)
+    partner_id = fields.Many2one(comodel_name="res.partner")
```
Comment (cesar-tecnativa) [inline sale_payment_sheet/wizards/sale_invoice_payment.py:35]: related ="invoice_ids.partner_id" ?

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

**Review by cesar-tecnativa — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**